internal/command/command.go

```diff
@@ -0,0 +1,221 @@
+	}
+
+	// create default grpc client
+	return gateway.NewGrpcGateway(host)
+}
+
+// create logger utility
+func createLogger() util.Logger {
+	// disable logging if we user want a specific format like JSON
+	//(more common they will not want also to have logs)
+	verbose := util.InfoLog
+	switch LogFlag {
+	case "none":
+		verbose = util.NoneLog
+	case "error":
+		verbose = util.ErrorLog
+	case "debug":
+		verbose = util.DebugLog
+	default:
+		verbose = util.InfoLog
+	}
+
+	if FormatFlag != "" {
+		verbose = util.NoneLog
+	}
+
+	return util.NewStdoutLogger(verbose)
+}
+
+// outputResult takes care of showing the result
+func formatResult(result Result) (string, error) {
+	if result == nil {
+		return "", fmt.Errorf("Missing")
+	}
+
+	if FilterFlag != "" {
+		var jsonResult map[string]interface{}
+		val, _ := json.Marshal(result.JSON())
+		err := json.Unmarshal(val, &jsonResult)
+		if err != nil {
+			return "", err
+		}
+
+		return fmt.Sprintf("%v", jsonResult[FilterFlag]), nil
+	}
+
+	switch FormatFlag {
+	case "json":
+		jsonRes, _ := json.Marshal(result.JSON())
+		return string(jsonRes), nil
+	case "inline":
+		return result.Oneliner(), nil
+	default:
+		return result.String(), nil
+	}
+}
+
+// outputResult to selected media
+func outputResult(result string) error {
+	if SaveFlag != "" {
+		af := afero.Afero{
+			Fs: afero.NewOsFs(),
+		}
+
+		fmt.Printf("💾 result saved to: %s \n", SaveFlag)
+		return af.WriteFile(SaveFlag, []byte(result), 0644)
+	}
+
+	// default normal output
+	fmt.Fprintf(os.Stdout, "%s\n", result)
+	return nil
+}
+
+// handleError handle errors
+func handleError(description string, err error) {
+	if err == nil {
+		return
+	}
+
+	// TODO: refactor this to better handle errors not by string matching
+	// handle rpc error
+	switch t := err.(type) {
+	case *client.RPCError:
+		fmt.Fprintf(os.Stderr, "❌  Grpc Error: %s \n", t.GRPCStatus().Err().Error())
+	default:
+		if strings.Contains(err.Error(), "transport:") {
+			fmt.Fprintf(os.Stderr, "❌  %s \n", strings.Split(err.Error(), "transport:")[1])
+		} else if strings.Contains(err.Error(), "NotFound desc =") {
+			fmt.Fprintf(os.Stderr, "❌  Not Found:%s \n", strings.Split(err.Error(), "NotFound desc =")[1])
+		} else if strings.Contains(err.Error(), "code = InvalidArgument desc = ") {
+			fmt.Fprintf(os.Stderr, "❌  Invalid argument: %s \n", strings.Split(err.Error(), "code = InvalidArgument desc = ")[1])
+		} else if strings.Contains(err.Error(), "invalid signature:") {
+			fmt.Fprintf(os.Stderr, "❌  Invalid signature: %s \n", strings.Split(err.Error(), "invalid signature:")[1])
+		} else {
+			fmt.Fprintf(os.Stderr, "❌  %s: %s", description, err)
+		}
+	}
+
+	fmt.Println()
+	os.Exit(1)
+}
+
+// bindFlags bind all the flags needed
+func bindFlags(command Command) {
+	err := command.GetFlags().
+		FromEnvironment(flow.EnvPrefix).
+		BindFlags(command.GetCmd().PersistentFlags()).
+		Parse()
+	if err != nil {
+		fmt.Println(err)
+	}
```

Comment:
In the version I suggested above, this just becomes:

```go
func bindFlags(command Command) {
	err := sconfig.New(command.Config).
		FromEnvironment(flow.EnvPrefix).
		BindFlags(command.Cmd.PersistentFlags()).
		Parse()
	if err != nil {
		fmt.Println(err)
	}
}

Reply:
Hmm I'm not sure I understand what you mean by suggested above also Command structure doesn't have a Config attr.